Python/Python.sublime-syntax

```diff
@@ -131,24 +133,18 @@ contexts:
     - match: ','
       scope: punctuation.separator.import-list.python
       push: expect-absolute-import
-    - match: (?=\bas\b)
-      set: import-alias-list
+    - include: imports-as
     - include: qualified-name
-    - match: (?=\S)
-      pop: true
+    - include: import-illegal-names
 
   imports-from-body:
     - meta_scope: meta.statement.import.python
     - meta_content_scope: meta.import-source.python
     - include: line-continuation-or-pop
     - match: (?=\bas\b)
```

Comment:
This seems to allow `from as foo` and `import as foo`. It's also broken in master but worth fixing here.

Reply:
I wouldn't call it broken in the first place, but rather a question of how to support/handle incomplete statements.

With regards to

https://github.com/sublimehq/Packages/blob/3783eebcc7416e08056df1c982a256094109b802/Python/syntax_test_python.py#L105-L114

... it is expected behavior.

We can either scope everything illegal, just because some part of a statement is missing, or we can silently accept it and try to recover as early as possible to maintain editing experience pleasent.

I've seen both strategies all over syntaxes so far. 

We also need to consider which of those illegal highlightings are better handled by linters. While illegals might be helpful in absence of linters, user experience might quickly become inconsistent if those are present, because static syntax definitions never can be as exact as linters.

Also seeing whole bunches of code in red or flickering single chars/words all the time while editing is a pain. It's quite normal for expressions and statements to be incomplete or invalid during authoring code, so I personally tend to argue for highlighting things illegal only, if it is sure to be wrong at a certain position not being likely to caused by incomplete statements. Otherwise just recover as early as possible continuing highlighting. We want to write syntax definitions, but no linters. Sophisticated illegal handling often introduces much more complexity.

That's also what happens in syntaxes which use multi-push often. If a token is not found at expected position, the context is just popped continuing with the next one. It would end up in exactly this behavior of supporting `import as foo`.

Reply:
Since `import  as foo` is something than can happen during typing, I agree with @deathaxe here in that it should not be marked as invalid. The flickering during typing is indeed annoying. However, we should still mark other keywords as illegal names in this place, e.g. `import from as foo`.

Reply:
I don’t agree with the idea that syntaxes should act as static analyzers. That’s what linters and language servers are for.